presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlan.java

```diff
@@ -29,44 +29,44 @@
 class RelationPlan
 {
     private final PlanNode root;
-    private final List<Symbol> outputSymbols;
+    private final List<Symbol> fieldMappings; // for each field in the relation, the corresponding symbol from "root"
     private final Scope scope;
 
-    public RelationPlan(PlanNode root, Scope scope, List<Symbol> outputSymbols)
+    public RelationPlan(PlanNode root, Scope scope, List<Symbol> fieldMappings)
     {
         requireNonNull(root, "root is null");
-        requireNonNull(outputSymbols, "outputSymbols is null");
+        requireNonNull(fieldMappings, "outputSymbols is null");
         requireNonNull(scope, "scope is null");
 
-        checkArgument(scope.getRelationType().getAllFieldCount() == outputSymbols.size(),
-                "Number of outputs (%s) doesn't match scope size (%s)", outputSymbols.size(), scope.getRelationType().getAllFieldCount());
+        checkArgument(scope.getRelationType().getAllFieldCount() == fieldMappings.size(),
+                "Number of outputs (%s) doesn't match scope size (%s)", fieldMappings.size(), scope.getRelationType().getAllFieldCount());
```

Comment:
can you please add a check that `root.getOuptutSymbols.size() == fieldMappings.size()`